Tests/Tests/TooltipDispatcherSpec.swift

```diff
@@ -187,6 +187,49 @@ class TooltipDispatcherSpec: QuickSpec {
                     expect(delegate.wasShouldShowCalled).toAfterTimeout(beFalse())
                 }
 
+                context("performing permission check") {
+
+                    context("and permission response perform ping are true") {
+
+                        it("will perform ping") {
+                            permissionService.shouldPerformPing = true
+                            let tooltip = TestHelpers.generateTooltip(id: "test", title: "[Tooltip][ctx] title", isTest: true)
+                            dispatcher.setNeedsDisplay(tooltip: tooltip)
+                            expect(delegate.wasPingCalled).toEventually(beTrue())
+                        }
+                    }
+
+                    context("and permission response perform ping are false") {
+
+                        it("will not perform ping") {
+                            permissionService.shouldPerformPing = false
+                            let tooltip = TestHelpers.generateTooltip(id: "test", title: "[Tooltip][ctx] title", isTest: true)
+                            dispatcher.setNeedsDisplay(tooltip: tooltip)
+                            expect(delegate.wasPingCalled).toEventually(beFalse())
+                        }
+                    }
+
+                    context("and permission response display are true") {
+
+                        it("will display the tooltip") {
+                            permissionService.shouldGrantPermission = true
+                            let tooltip = TestHelpers.generateTooltip(id: "test", title: "[Tooltip][ctx] title", isTest: true)
+                            dispatcher.setNeedsDisplay(tooltip: tooltip)
+                            expect(router.displayedTooltips).toEventuallyNot(beEmpty())
+                        }
+                    }
+
+                    context("and permission response display are false") {
+
+                        it("will not display the tooltip") {
+                            permissionService.shouldGrantPermission = false
+                            let tooltip = TestHelpers.generateTooltip(id: "test", title: "[Tooltip][ctx] title", isTest: true)
+                            dispatcher.setNeedsDisplay(tooltip: tooltip)
+                            expect(router.displayedTooltips).toEventually(beEmpty())
```

Comment:
since `router.displayedTooltips` is initially empty, this expectation will be fulfilled instantly. In this case you should use `.toAfterTimeout(beEmpty())` to check if `displayedTooltips` is empty after 1 second

Reply:
Fixed here [4fcfb1f](https://github.com/rakutentech/ios-inappmessaging/pull/252/commits/612526bdf5401f2de80f19ed29c9a36e74fcfb1f).